eth/handler.go

```diff
@@ -286,7 +286,7 @@ func (pm *ProtocolManager) handle(p *peer) error {
 		}
 		if !fork.RequiredHash.IsEmpty() {
 			// Request the peer's fork block header for extra-dat
-			if err := p.RequestHeadersByNumber(fork.Block.Uint64(), 1, 0, false); err != nil {
+			if err := p.RequestHeadersByNumber(fork.Block.Uint64(), 1, 0, false); err != nil && err != core.ErrHashEmpty {
```

Comment:
Okay I think `ErrHashEmpty` would be reasonable then.

However, in this line, if the peer sends back a header empty hash, then isn't it considered that this peer is misbehaving and should be dropped? If this line makes any test fails, then I think we should probably fix the test instead?